### PR TITLE
Enable ParallelExecutor in SSD-MobileNet and Refine code.

### PR DESCRIPTION
**Pull request by qingqing01**

### CODE REVIEW

File: fluid/object_detection/train.py [hunk 1]
```diff
@@ -12,46 +12,35 @@
...
+
+def parallel_do(args,
+                train_file_list,
+                val_file_list,
+                data_args,
+                learning_rate,
+                batch_size,
+                num_passes,
+                model_save_dir,
+                pretrained_model=None):
```
Comment (chengduoZH) [inline fluid/object_detection/train.py:44]: `parallel_do` need have an argument to define using `nccl` or not. The default value does not use `nccl`.

File: fluid/object_detection/train.py [hunk 2]
```diff
@@ -164,29 +155,148 @@ def test(pass_id):
...
+    place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+    exe.run(fluid.default_startup_program())
+
+    if pretrained_model:
+        def if_exist(var):
+            return os.path.exists(os.path.join(pretrained_model, var.name))
+        fluid.io.load_vars(exe, pretrained_model, predicate=if_exist)
+
+    train_exe = fluid.ParallelExecutor(use_cuda=True, loss_name=loss.name)
```
Comment (chengduoZH) [inline fluid/object_detection/train.py]: line 218 and line 227 is not consistent, currently `ParallelExecutor` only allows to use GPU.

File: fluid/object_detection/train.py [hunk 3]
```diff
@@ -164,29 +155,148 @@ def test(pass_id):
...
+        place=place, feed_list=[image, gt_box, gt_label, difficult])
+
+    def test(pass_id):
+        _, accum_map = map_eval.get_map_var()
+        map_eval.reset(exe)
+        test_map = None
+        for _, data in enumerate(test_reader()):
+            test_map = exe.run(test_program,
+                               feed=feeder.feed(data),
+                               fetch_list=[accum_map])
```
Comment (chengduoZH) [inline fluid/object_detection/train.py:245]: `exe.run` is not the format of `parallel_exe`.

File: fluid/object_detection/train.py (hunk 1 above)
Comment (qingqing01) [reply fluid/object_detection/train.py:44]: Done.

File: fluid/object_detection/train.py (hunk 3 above)
Comment (qingqing01) [reply fluid/object_detection/train.py:245]: The testing use Executor, not ParallelExecutor.

File: fluid/object_detection/train.py (hunk 2 above)
Comment (qingqing01) [reply fluid/object_detection/train.py]: Here change `use_cuda=arg.use_gpu`. Thanks!

**Review by chengduoZH — APPROVED** [review]
LGTM

**Pull request merged**